src/Apps/W1/Subscription Billing/App/Customer Contracts/Pages/ExtendContract.Page.al

```diff
@@ -166,6 +166,29 @@ page 8002 "Extend Contract"
                     ToolTip = 'Specifies the item description for the Subscription Item to be created.';
                     Editable = false;
                 }
+                field(VariantCode; VariantCode)
+                {
+                    Caption = 'Variant Code';
+                    ToolTip = 'Specifies the variant of the item.';
+                    Visible = UsageDataSupplierNo = '';
```

Comment:
This looks like a conditional visibility based on page field that can be changed.
If this should work like expected then you would need to wrap this field into a group and hide the group dynamicly instead.